### PR TITLE
Colin Narans

### PR DESCRIPTION
**Pull request by cnarans**

### CODE REVIEW

File: lib/hashes.md [hunk 1]
```diff
@@ -5,40 +5,40 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 
 # What is a Hash?
 
-Replace me with your response...
+A Hash is a collection of data elements that are indexed by a key that is another data element.
```
Comment (halfghaninne) [inline lib/hashes.md:8]: I don't think there is anything incorrect about your answer, but I'd like to encourage you to be a little more precise. 

For some hash:
`hash = {"key1" => "value1", "key2" => "value2", "key3" => "value3"}`

The hash is a _collection_ of _key-value pairs_, indexed on the _keys_. Both the _key_ and _value_ can be any kind of Ruby Object.

File: lib/hashes.md [hunk 2]
```diff
@@ -5,40 +5,40 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
...
 
 # How are Hashes and Arrays similar? How are they different?
 
-Replace me with your response...
+Both are collections of data elements, but Arrays are indexed by sequential integers, while Hashes are indexed by other data elements.
 
 # How do you retrieve a particular value from a Hash?
 
-Replace me with your response...
+By using the data element associated with that value as the index key.
```
Comment (halfghaninne) [inline lib/hashes.md:20]: It's hard for me to picture what you're trying to convey here. Using an example might help clarify.

Comment (cnarans) [inline lib/hashes.md:20]: I think I'm probably not using the correct terminology.  By "data element associated with that value" I just meant whatever you're using for the key.  In your hash example above, you would use hash["key1"] to retrieve "value1", hash["key2"] to retrieve "value2" etc, but the key and value can be any kind of Ruby object, as you said.

**Comment by halfghaninne** [comment]
Awesome! That clarification helped. Nice work.

**Pull request closed without merging**